src/index.ts

```diff
@@ -45,23 +47,51 @@ async function main() {
   });
   Prometheus.collectDefaultMetrics({ register });
 
-  const client = new Client({
-    url: conf.mercuryGraphQL,
-    exchanges: [fetchExchange],
-    fetchOptions: () => {
-      return {
-        headers: { authorization: `Bearer ${conf.mercuryKey}` },
-      };
-    },
-  });
-  // we need a second client because the authenticate muation does not ignore the current jwt
-  const renewClient = new Client({
-    url: conf.mercuryGraphQL,
-    exchanges: [fetchExchange],
-  });
+  const graphQlEndpoints = {
+    TESTNET: conf.mercuryGraphQLTestnet,
+    PUBLIC: conf.mercuryGraphQLPubnet,
+  };
+
+  const backends = {
+    TESTNET: conf.mercuryBackendTestnet,
+    PUBLIC: conf.mercuryBackendPubnet,
+  };
+
+  function renewClientMaker(network: NetworkNames) {
+    if (!hasIndexerSupport(network)) {
+      throw new Error(`network not currently supported: ${network}`);
```

Comment:
you have similar errors in some different areas. Would it be better to vary these error messages so we can more easily trace back logs to where they were thrown?

Reply:
Would a line number give you enough context for this? I think we could give them each a different message for that but probably cleaner to align error messages to error types and use stack trace or line number to know where it threw?

Reply:
Yup, I think line number is fine! Just something to trace back